### PR TITLE
Add a new rake task to redirect some old unpublished statistics announcements

### PR DESCRIPTION
**Pull request by minhngocd**
Unfortunately Statistics Announcements behave differently to other Document Types.

Also changed aborts to next within this rake task file. The assertions within tests are skipped if abort is run in the rake task.

https://trello.com/c/VNP40fAr/3276-superseded-by-another-page-unpublish-content-request

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: lib/tasks/publishing_api.rake [hunk 1]
```diff
@@ -133,7 +133,10 @@ namespace :publishing_api do
       args.with_defaults(locale: "en")
 
       document = Document.find_by(content_id: args[:content_id])
-      abort "Document with this content ID exists: #{document}" if document
+      if document
+        puts "Document with this content ID exists: #{document}"
+        next
+      end
```
Comment (ryanb-gds) [inline lib/tasks/publishing_api.rake:139]: This probably ought to be a separate commit really

**Review by ryanb-gds — APPROVED** [review]
Seems like we're missing a feature here of being able to re-redirect unpublished stats announcements, but this will do for now. Do we have a Trello card of recurring support ticlets we can log this on to see if it comes up again?

File: lib/tasks/publishing_api.rake [hunk 2]
```diff
@@ -146,6 +149,42 @@ namespace :publishing_api do
...
+  task :redirect_unpublished_statistics_announcement, %i[slug alternative_url locale] => :environment do |_, args|
+    args.with_defaults(locale: "en")
+
+    results = StatisticsAnnouncement.unscoped.where(slug: args[:slug])
+    if results.empty?
+      puts "Could not find Statistics Announcement with slug #{args[:slug]}"
+      next
+    end
+    if results.count > 1
+      puts "More than one Statistics Announcement with slug #{args[:slug]}"
```
[the remark this answers is not in the document]
Comment (minhngocd) [reply lib/tasks/publishing_api.rake:164]: Well - i'm guessing because they are "unpublished" it's highly likely - as publishers might create something `slug-123` and decide to unpublish it, it disappears from Whitehall, and then they change their mind and create a new one with the same name. I don't know if Statistics Announcement slug generation would take into account `unpublished` ones for slug check as it's not part of the default scope. 

In either case, there is no uniqueness constraint on the DB and the tests did allow me to create multiple unpublishings of the same slug so... 🙈

File: lib/tasks/publishing_api.rake (hunk 1 above)
Comment (minhngocd) [reply lib/tasks/publishing_api.rake:139]: I can split that out - no probs 👍

**Comment by minhngocd** [comment]
> Seems like we're missing a feature here of being able to re-redirect unpublished stats announcements, but this will do for now. Do we have a Trello card of recurring support tickets we can log this on to see if it comes up again?

Yes agreed - and no I don't think we do have a Trello card. I could pop one into the backlog, but not sure how to make it more visible so people know to log onto that ticket.

File: lib/tasks/publishing_api.rake (hunk 2 above)
[the remark this answers is not in the document]
Comment (minhngocd) [reply lib/tasks/publishing_api.rake:164]: I'll make it clearer with the message

[the remark this answers is not in the document]
Comment (ryanb-gds) [reply lib/tasks/publishing_api.rake:164]: Oh, I see. Thanks for explaining

**Comment by ryanb-gds** [comment]
> > Seems like we're missing a feature here of being able to re-redirect unpublished stats announcements, but this will do for now. Do we have a Trello card of recurring support tickets we can log this on to see if it comes up again?
> 
> Yes agreed - and no I don't think we do have a Trello card. I could pop one into the backlog, but not sure how to make it more visible so people know to log onto that ticket.

Maybe add to the retro board for January that James posted on the Whitehall channel the other day?
